* Don't look in /usr/local unconditionally.  In Prefix we don't like
  that, as we have all the cool libs and binaries in our prefix.

--- src/configure.in
+++ src/configure.in
@@ -186,8 +186,22 @@
 dnl Add /usr/local/lib to $LDFLAGS and /usr/local/include to CFLAGS.
 dnl Only when the directory exists and it wasn't there yet.
 dnl For gcc don't do this when it is already in the default search path.
+AC_MSG_CHECKING(for local dir)
+local_dir=/usr/local
 have_local_include=''
 have_local_lib=''
+AC_ARG_WITH(local-dir, [], [
+  local_dir="$withval"
+  case "$withval" in
+  */*) ;;
+  no)
+    # avoid adding local dir to LDFLAGS and CPPFLAGS
+    have_local_dir=yes
+    have_local_lib=yes
+    ;;
+  *) AC_MSG_ERROR(must pass path argument to --with-local-dir) ;;
+  esac
+], [
 if test "$GCC" = yes; then
   echo 'void f(){}' > conftest.c
   dnl -no-cpp-precomp is needed for OS X 10.2 (Ben Fowler)
@@ -195,18 +209,20 @@
   have_local_lib=`${CC-cc} -c -v conftest.c 2>&1 | grep '/usr/local/lib'`
   rm -f conftest.c conftest.o
 fi
-if test -z "$have_local_lib" -a -d /usr/local/lib; then
-  tt=`echo "$LDFLAGS" | sed -e 's+-L/usr/local/lib ++g' -e 's+-L/usr/local/lib$++g'`
+])
+if test -z "$have_local_lib" -a -d "${local_dir}"/lib; then
+  tt=`echo "$LDFLAGS" | sed -e "s+-L${local_dir}lib ++g" -e "s+-L${local_dir}/lib$++g"`
   if test "$tt" = "$LDFLAGS"; then
-    LDFLAGS="$LDFLAGS -L/usr/local/lib"
+    LDFLAGS="$LDFLAGS -L${local_dir}/lib"
   fi
 fi
-if test -z "$have_local_include" -a -d /usr/local/include; then
-  tt=`echo "$CPPFLAGS" | sed -e 's+-I/usr/local/include ++g' -e 's+-I/usr/local/include$++g'`
+if test -z "$have_local_include" -a -d "${local_dir}"/include; then
+  tt=`echo "$CPPFLAGS" | sed -e "s+-I${local_dir}/include ++g" -e "s+-I${local_dir}/include$++g"`
   if test "$tt" = "$CPPFLAGS"; then
-    CPPFLAGS="$CPPFLAGS -I/usr/local/include"
+    CPPFLAGS="$CPPFLAGS -I${local_dir}/include"
   fi
 fi
+AC_MSG_RESULT($local_dir)
 
 AC_MSG_CHECKING(--with-vim-name argument)
 AC_ARG_WITH(vim-name, [  --with-vim-name=NAME    what to call the Vim executable],
